Sources/Algorithms/Chunked.swift

```diff
@@ -246,3 +246,136 @@ extension Collection {
+  }
+}
+
+extension ChunkedCollection: Collection {
+  public struct Index {
+    public let base: Base.Index
+    
+    @usableFromInline
+    init(_base: Base.Index) {
+      self.base = _base
+    }
+  }
+  
+  public var startIndex: Index { Index(_base: base.startIndex) }
+  public var endIndex: Index { Index(_base: base.endIndex) }
+  
+  public subscript(i: Index) -> Element {
+    base[i.base..<index(after: i).base]
+  }
+  
+  @inlinable
+  public func index(after i: Index) -> Index {
+    let baseIdx = base.index(
+      i.base, offsetBy: _chunkCount, limitedBy: base.endIndex)
+    return Index(_base: baseIdx ?? base.endIndex)
+  }
+}
+
+extension ChunkedCollection.Index: Comparable {
+  @inlinable
+  public static func < (lhs: ChunkedCollection.Index,
+                        rhs: ChunkedCollection.Index) -> Bool {
+    lhs.base < rhs.base
+  }
+}
+
+extension ChunkedCollection:
+  BidirectionalCollection, RandomAccessCollection
+where Base: RandomAccessCollection {
+  @inlinable
+  public func index(before i: Index) -> Index {
+    if i.base == base.endIndex {
+      let remainder = base.count%_chunkCount
+      if remainder != 0 {
+        return Index(_base: base.index(i.base, offsetBy: -remainder))
+      }
+    }
+    return Index(_base: base.index(i.base, offsetBy: -_chunkCount))
+  }
+  
+  @inlinable
+  public func distance(from start: Index, to end: Index) -> Int {
+    let distance = base.distance(from: start.base, to: end.base)
+    let (quotient, remainder) =
+      distance.quotientAndRemainder(dividingBy: _chunkCount)
+    return quotient + remainder
+  }
+  
+  @inlinable
+  public func index(_ i: Index, offsetBy n: Int) -> Index {
+    guard n != 0 else { return i }
+    
+    let bound = n > 0 ? base.endIndex : base.startIndex
+    return Index(_base:
+      base.index(i.base, offsetBy: n * _chunkCount, limitedBy: bound) ?? bound
+    )
+  }
+  
+  @inlinable
+  public var count: Int {
+    return base.count
+      .isMultiple(of: _chunkCount) ? base.count/_chunkCount
+                                   : base.count/_chunkCount + 1
+  }
+}
+
+extension Collection {
+  /// Returns a `ChunkedCollection<Self>` view presenting the elements
+  /// in chunks with count of the given size parameter.
+  ///
+  /// - Parameter size: The size of the chunks. If the size parameter
+  ///   is evenly divided by the count of the base `Collection` all the
+  ///   chunks will have the count equals to size.
+  ///   Otherwise, the last chunk will contain the remaining elements.
+  ///
+  ///     let c = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+  ///     print(c.chunks(ofCount: 5).map(Array.init))
+  ///     // [[1, 2, 3, 4, 5], [6, 7, 8, 9, 10]]
+  ///
+  ///     print(c.chunks(ofCount: 3).map(Array.init))
+  ///     // [[1, 2, 3], [4, 5, 6], [7, 8, 9], [10]]
+  ///
+  /// - Complexity: O(1)
+  @inlinable
+  public func chunks(ofCount count: Int) -> ChunkedCollection<Self> {
+    assert(count > 0, " Cannot chunk with size <= 0!")
```

Comment:
Maybe this should be a precondition, so it also traps in release builds?
```suggestion
    precondition(count > 0, " Cannot chunk with count <= 0!")
```